asio/CMakeLists.txt

```diff
@@ -0,0 +1,240 @@
+# Minimum supported CMake version
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+if(CONAN_COMPILER)
+
+  # ==========================================================
+  project(asio VERSION ${CONAN_PACKAGE_VERSION} LANGUAGES CXX)
+  # ==========================================================
+
+  include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+  conan_basic_setup(TARGETS)    # this set CONAN_TARGETS
+
+else()
+
+  # use ccache and clang++ if found
+  find_program(CCACHE_EXECUTABLE "ccache" HINTS /usr/local/bin /opt/local/bin)
+  if(CCACHE_EXECUTABLE)
+    message(STATUS "use ccache")
+    set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_EXECUTABLE}" CACHE PATH "ccache" FORCE)
+    set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_EXECUTABLE}" CACHE PATH "ccache" FORCE)
+  endif()
+
+  # ========================================
+  project(asio VERSION 1.14.1 LANGUAGES CXX)
+  # ========================================
+
+endif()
+
+#----------------------------------------------------------
+# Compiler config
+#----------------------------------------------------------
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+message(STATUS "USING: CMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}")
+message(STATUS "USING: CMAKE_CXX_STANDARD_REQUIRED=${CMAKE_CXX_STANDARD_REQUIRED}")
+message(STATUS "USING: CMAKE_CXX_EXTENSIONS=${CMAKE_CXX_EXTENSIONS}")
```

Comment:
Don't use `CMAKE_CXX_STANDARD` in project CMLs, it's supposed to be used to override build requirements from the CLI. Use [`target_compile_features`](https://cmake.org/cmake/help/latest/command/target_compile_features.html) to specify a minimum requirement. Let the user decide how they want to build.